lib/Model/IMAPMessage.php

```diff
@@ -153,6 +153,12 @@ public function setFlags(array $flags) {
 		throw new Exception('Not implemented');
 	}
 
+	/**
+	 */
+	public function getRawMessage() {
```

Comment:
this should be a method of https://github.com/nextcloud/mail/blob/master/lib/IMAP/MessageMapper.php. See https://github.com/nextcloud/mail/blob/ec940371233c2c3630d5fb48e65e9e75203cbbc1/lib/IMAP/MessageMapper.php#L244-L309 for reference

Reply:
I shouldn't be using `$this->fetch->getFullMsg()` here. Is this also what you mean?

Reply:
Yes, please don't add any more complexity to this class. This coding style is not testable. I'm migrating these methods into simple, testable service methods.

You can use `getFullMsg` in `MessageMapper`, though :)